### PR TITLE
Ticket Details Page

### PR DESCRIPTION
**Pull request by sreenidhi3**
- edit ticket API integrated
- reopen ticket API integrated
- timeline to display activities

### CODE REVIEW

File: src/modules/Category/category.hook.ts [hunk 1]
```diff
@@ -39,6 +40,7 @@ export const useCategories = (dept_id) => {
     [API_CONSTANTS.CATEGORY_LIST, dept_id],
     () => getCategoriesList(dept_id),
     {
+      enabled: !!dept_id,
       onError: () => {
```
Comment (ayushjnv1) [inline src/modules/Category/category.hook.ts:44]: same hear or also if it is not boolean value boolean(dept_id) it works

File: src/modules/Ticket/ticket.hook.ts [hunk 2]
```diff
@@ -11,12 +11,10 @@ import API_CONSTANTS from 'hooks/constants';
...
 
 export const useCreateTicket = () => {
-  // const queryClient = useQueryClient();
   const { mutate, data, isLoading, error } = useMutation(
     (payload: ICreateTicketPayload) => postCreateTicket(payload),
     {
       onSuccess: (res) => {
-        // queryClient.invalidateQueries([API_CONSTANTS.TICKET]);
         toast.success(res?.data?.message);
       },
```
Comment (ayushjnv1) [inline src/modules/Ticket/ticket.hook.ts:19]: if this message is undefined then blank toast is generated

File: src/modules/details/details.hook.ts [hunk 3]
```diff
@@ -1,32 +1,72 @@
...
+    }: {
+      id: number;
+      ticket_result: { ticket_result: IReopenTicketPayload };
+    }) => putReopenTicket({ id, ticket_result }),
     {
-      onError: (err) => {
+      onSuccess: (res) => {
+        toast.success(res?.data?.message);
+        queryClient.invalidateQueries([API_CONSTANTS.DETAILS_SPECEFIC]);
+      },
```
Comment (ayushjnv1) [inline src/modules/details/details.hook.ts:46]: toast me default value honi chaiye nahi to undefine ke time empty toast generate hoga

**Comment by sreenidhi3** [comment]
Made the requested changes. Please check.

**Pull request merged**